homeassistant/components/soma/sensor.py

```diff
@@ -43,11 +43,6 @@ def native_value(self):
     async def async_update(self) -> None:
         """Update the sensor with the latest data."""
         response = await self.get_battery_level_from_api()
-
-        # https://support.somasmarthome.com/hc/en-us/articles/360026064234-HTTP-API
-        # battery_level response is expected to be min = 360, max 410 for
-        # 0-100% levels above 410 are consider 100% and below 360, 0% as the
-        # device considers 360 the minimum to move the motor.
-        _battery = round(2 * (response["battery_level"] - 360))
+        _battery = response["battery_percentage"]
```

Comment:
Why we can't support both ways?

Assuming only devices with the new firmware have `battery_percentage` in the response, we could do something like this:

```suggestion 
        _battery = response.get("battery_percentage")
        if _battery is None:
            # https://support.somasmarthome.com/hc/en-us/articles/360026064234-HTTP-API
            # battery_level response is expected to be min = 360, max 410 for
            # 0-100% levels above 410 are consider 100% and below 360, 0% as the
            # device considers 360 the minimum to move the motor.
            _battery = round(2 * (response["battery_level"] - 360))
```